test/test_optim.py

```diff
@@ -230,6 +230,15 @@ def _test_basic_cases(self, constructor, ignore_multidevice=False):
             constructor
         )
 
+    def _test_error_scenarios(self, constructor, error_msg):
+        with self.assertRaisesRegexp(ValueError, error_msg):
+            self._test_basic_cases_template(
+                torch.randn(10, 5),
+                torch.randn(10),
+                torch.randn(5),
+                constructor
+            )
```

Comment:
There’s no reason to run the whole test if you expect the constructor to fail. Just assert that it fails upon construction